pkg/virt-api/webhooks/validating-webhook/admitters/vmrestore-admitter.go

```diff
@@ -162,6 +167,54 @@ func (admitter *VMRestoreAdmitter) Admit(ctx context.Context, ar *admissionv1.Ad
 	return &reviewResponse
 }
 
+func (admitter *VMRestoreAdmitter) validateTargetVM(ctx context.Context, field *k8sfield.Path, vmRestore *snapshotv1.VirtualMachineRestore) (causes []metav1.StatusCause, err error) {
+	targetName := vmRestore.Spec.Target.Name
+	namespace := vmRestore.Namespace
+
+	causes = admitter.validatePatches(vmRestore.Spec.Patches, field.Child("patches"))
+
+	vmSnapshot, err := admitter.Client.VirtualMachineSnapshot(namespace).Get(ctx, vmRestore.Spec.VirtualMachineSnapshotName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	target, err := admitter.Client.VirtualMachine(namespace).Get(ctx, targetName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	}
+
+	sourceTargetVmsAreDifferent := errors.IsNotFound(err) || (vmSnapshot.Status.SourceUID != nil && target.UID != *vmSnapshot.Status.SourceUID)
+	if sourceTargetVmsAreDifferent {
+		contentName := vmSnapshot.Status.VirtualMachineSnapshotContentName
+		if contentName == nil {
+			return nil, fmt.Errorf("snapshot content name is nil in vmSnapshot status")
+		}
+
+		vmSnapshotContent, err := admitter.Client.VirtualMachineSnapshotContent(namespace).Get(ctx, *contentName, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		snapshotVM := vmSnapshotContent.Spec.Source.VirtualMachine
+		if snapshotVM == nil {
+			return nil, fmt.Errorf("unexpected snapshot source")
+		}
+
+		if backendstorage.IsBackendStorageNeededForVMI(&snapshotVM.Spec.Template.Spec) {
+			causes = append(causes, metav1.StatusCause{
+				Type:    metav1.CauseTypeFieldValueInvalid,
+				Message: "Restore to a different VM not supported when using backend storage",
+				Field:   field.String(),
```

Comment:
Could you add a few works in the commit message about the reasoning we block this?

Reply:
Sure

Reply:
ACK